### PR TITLE
Work on ISLANODRA-2421 - multiple Payload-Oxum tags not allowed.

### PR DESCRIPTION
**Pull request by mjordan**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2421

# What does this Pull Request do?

Fixes bug where a Payload-Oxum tag is added to bag-info.txt is added for each enabled plugin. Only one Payload-Oxum tag must be added to bag-info.txt.

# What's new?

The call to the function that generated Payload-Oxum was within the loop that iterated over all enabled plugins. It now follows that loop.

# How should this be tested?

1. Check out this branch.
1. In the Bagit config settings, enable the Payload-Oxum tag.
1. Enable more than on plugin.
1. Generate a Bag for an object.
1. Look inside that Bag's bag-info.txt. There should only be one Payload-Oxum tag.

# Interested parties
@Natkeeran, @MarcusBarnes

### CODE REVIEW

**Comment by mjordan** [comment]
Hold off merging - I have to make sure this bug doesn't apply to collection Bags as well.

**Comment by mjordan** [comment]
It does not apply to collection Bags. Please test.

**Comment by MarcusBarnes** [comment]
@mjordan I tested this PR with the `plugin_object_foxml` and `plugin_object_premis` plugins enabled.  What I get in the bagit-info.txt is one Payload-Oxum line, but it only reports the payload of the last processed plugin.  In my case,  the Payload-Oxum line is something like "Payload-Oxum: [size_of_PREMIS.xml].1".  The expected behaviour is to have the Payload-oxum metadata line read something like: "Payload-Oxum: [Size of foxml.xml + size of PREMIS.xml].2".

**Comment by mjordan** [comment]
OK, thanks, I'll take another look, probably won't be for a day or two.

**Comment by MarcusBarnes** [comment]
I'll mention that when using only the `plugin_object_ds_basic` plugin, the Payload-Oxum line gives the correct values, namely the total size of all the files in the data/ folder and the total number of files:  "Payload-Oxum: [total size of all files].[number of files]".

**Comment by mjordan** [comment]
@MarcusBarnes please test again. This is not yet ready to merge. I still need to confirm that the Payload-Oxum for collection Bags is still working.

**Comment by MarcusBarnes** [comment]
@mjordan Thanks.  I'll give it another test.

**Comment by mjordan** [comment]
Thanks. This change appears to have had no effect on collection Bags, so if the original problem is resolved, it's ready to merge.

**Comment by mjordan** [comment]
Again with Travis failing on the PHP 5.3.3 / Fedora 3.8.1 combination with `[Composer\Exception \NoSslException The openssl extension is required for SSL/TLS protection but is not available. If you can not enable the openssl extension, you can disable this error, at your own risk, by setting the 'disable-tls' option to true.`

**Comment by MarcusBarnes** [comment]
@mjordan I tested again and I get the results I expect when using multiple plugins (though I haven't tested each and every permutation of plugins).  Thanks!

**Comment by mjordan** [comment]
@MarcusBarnes thanks. The specific combination of plugins shouldn't matter since the problem was the location of the code that generates the sum in relation to the `foreach` loop that iterates over enabled plugins (in other words, it was a logic problem, not a plugin problem).

**Comment by MarcusBarnes** [comment]
@mjordan Regarding the Travis build failure, see https://github.com/Islandora/islandora_scholar/blob/7.x/.travis.yml#L36

**Comment by mjordan** [comment]
@MarcusBarnes The workaround @whikloj uses in https://github.com/Islandora/islandora_solution_pack_collection/pull/216 eliminates PHP 5.3.3 instead of just the TLS failure. I wonder if that more general fix is a better choice?

**Comment by MarcusBarnes** [comment]
If the TLS failure workaround allows the rest of the PHP 5.3.3. + Fedora 3.8.1 build to continue so that all the other checks could go ahead for that configuration that would be the approach I'd recommend, otherwise you'll see a passing Travis build without knowing if work was compatible with the relevant PHP 5.3.3 + Fedora 3.x builds.    If that workaround doesn't work, the approach @whikloj suggested would allow the PRs to be merged while investigating the specific PHP 5.3.3 build issues.  But I'd document such as workaround as an issue - that you had to create a workaround for this version of PHP (known to be problematic, but still supported because of the handful of instances still using it in the community).

**Comment by mjordan** [comment]
Hm, that approach didn't resolve the TLS issue, and it also failed on one of the PHP 5.4 builds with "The command "phpenv global 5.4" failed and exited with 1 during ." I'll try the other approach to see what happens (although that approach will not address the 5.4 failure).

**Review by MarcusBarnes — APPROVED** [review]

**Comment by mjordan** [comment]
@MarcusBarnes can you add a review so we can merge this (assuming it is ready to merge)?

**Review by MarcusBarnes — APPROVED** [review]

**Comment by MarcusBarnes** [comment]
I think my review doesn't apply as I don't have write access to this repo.  A review is required by "reviewers with write access."

**Comment by mjordan** [comment]
OK, thanks.

@Islandora/7-x-1-x-committers , lil help?

**Review by bondjimbond — APPROVED** [review]
Looks good.

File: islandora_bagit.module
Comment (DiegoPino) [inline islandora_bagit.module:1140]: @mjordan question. is 0.0 a valid Payload-oxum? If you have no sources (edge case?) not even sure if possible.
Also filesize($file) returns FALSE if there is an issue (disk full, can't read, who knows). You would probably want to cast it to an (int) in that case

Comment (mjordan) [reply islandora_bagit.module:1140]: @DiegoPino the Bagit spec doesn't prohibit an empty `/data` directory, and defines the Payload-Oxum as "the total number of octets (8-bit bytes) across  all payload file content and [...] the total number of payload files" so I'd say it's allowed. I haven't tested this reading of the spec against any BagIt validators however.

Thanks for finding the `filesize()` omission, I'll fix that.

Comment (mjordan) [reply islandora_bagit.module:1140]: Added in 6fc5f09a22b937ecf78f6120bb7ab1514fe75148.

**Comment by DiegoPino** [comment]
Perfect. Thanks. Since this was already approved by @bondjimbond should be ready to merge in 12 hours.

**Comment by mjordan** [comment]
bump on merging.....

**Pull request merged**
